### PR TITLE
Disable default instance delete

### LINKED ISSUE #11214: Users can try to edit/delete Instance group "default" via UI
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](http://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I am not entitled to status updates or other assurances.

### Summary

Users can try to edit/delete Instance group "default" via UI, however, this action is not allowed by API.

### AWX version

devel

### Installation method

openshift

### Modifications

no

### Ansible version

_No response_

### Operating system

_No response_

### Web browser

Chrome

### Steps to reproduce

1. Install controller using
 - 1 control only node
 - 2 execution node

2. Go to instance group "default"
3. try to edit the name or even delete this group

### Expected results

1. The name field should be disabled as well as controlplane group
2. The delete button should not exist because the API will always return 403 if the user attempt to delete

### Actual results

unnamed (1)

unnamed (2)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
Resolves #11214 
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2021-10-07 at 11 48 45 AM
Screen Shot 2021-10-07 at 11 48 37 AM

### CODE REVIEW

**Comment by nixocio** [comment]
@AlexSCorey it seems like there is an extra commit on your branch - commit that was already merged.

image

File: awx/ui/src/screens/InstanceGroup/shared/InstanceGroupForm.test.js [hunk 1]
```diff
@@ -116,4 +119,44 @@ describe('<InstanceGroupForm/>', () => {
     wrapper.find('button[aria-label="Cancel"]').invoke('onClick')();
     expect(onCancel).toBeCalled();
   });
+
+  test('Name field should be disabled, defualt', async () => {
```
Comment (nixocio) [inline awx/ui/src/screens/InstanceGroup/shared/InstanceGroupForm.test.js]:
```suggestion
  test('Name field should be disabled, default', async () => {
```

File: awx/ui/src/screens/InstanceGroup/shared/InstanceGroupForm.test.js [hunk 2]
```diff
@@ -116,4 +119,44 @@ describe('<InstanceGroupForm/>', () => {
...
+      );
+    });
+    expect(
+      defaultInstanceGroupWrapper
+        .find('TextInput[name="name"]')
+        .prop('isDisabled')
+    ).toBe(true);
+  });
+
+  test('Name field should be disabled, controleplane', async () => {
```
Comment (nixocio) [inline awx/ui/src/screens/InstanceGroup/shared/InstanceGroupForm.test.js]:
```suggestion
  test('Name field should be disabled, controlplane', async () => {
```

**Review by nixocio — CHANGES_REQUESTED** [review]
Good job @AlexSCorey .

Code changes looks good. 

Minor suggestion to add a tooltip explaining why the name change is disabled. 

As an example:

image

Also, are we going to hide delete button here?

image



We need to fix the extra commit before merging.

**Comment by nixocio** [comment]
> @AlexSCorey it seems like there is an extra commit on your branch - commit that was already merged.
> 
> image


@AlexSCorey There are just 2 commits now, but the changes related to the third one remains.

**Review by tiagodread — APPROVED** [review]
This looks good from QE standpoint

**Review by nixocio — APPROVED** [review]

**Pull request merged**